### PR TITLE
Make token expiration configurable.

### PR DESCRIPTION
**Pull request by bjhaid**
Requested overrides can only be short lived tokens and cannot have a
lifespan longer than the configured token expiration.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -236,6 +238,19 @@ func authenticateHandler(c *Config) func(http.ResponseWriter, *http.Request) {
 func tokenHandler(c *Config) func(w http.ResponseWriter, r *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 		username, password, ok := r.BasicAuth()
+		tokenExpiresIn := c.TokenExpiresIn
+		if v := r.URL.Query().Get("token-expires-in"); v != "" {
+			if t, err := strconv.Atoi(v); err != nil {
+				glog.Errorf("Got a non int token expiry: %s", v)
```
Comment (2rs2ts) [inline main.go]: Just my $0.02 but I would 400 in this case

File: main.go [hunk 2]
```diff
@@ -236,6 +238,19 @@ func authenticateHandler(c *Config) func(http.ResponseWriter, *http.Request) {
...
+			} else {
+				if t > tokenExpiresIn {
+					glog.V(2).Infof("Ignoring token expiry %d, which is greater than" +
```
Comment (2rs2ts) [inline main.go]: same here, I would 400

Comment (bjhaid) [reply main.go]: addressed

File: main.go (hunk 1 above)
Comment (bjhaid) [reply main.go]: addressed

**Pull request merged**